sdk/go/common/util/executable/executable.go

```diff
@@ -32,18 +32,27 @@ func FindExecutable(program string) (string, error) {
 		return cwdProgram, nil
 	}
 
-	// look in $GOPATH/bin
+	// look in potentials $GOPATH/bin
 	if goPath := os.Getenv("GOPATH"); len(goPath) > 0 {
-		goPathProgram := filepath.Join(goPath, "bin", program)
-		fileInfo, err := os.Stat(goPathProgram)
-		if err != nil {
-			if !os.IsNotExist(err) {
-				return "", errors.Wrapf(err, "unable to find program in %q", goPathProgram)
+		// getPotentialPaths will return paths where to look.
+		// Because the GOPATH can take the form of multiple paths (e.g: GOPATH="home/user/go:/usr/local/go")
+		// we need to split the GOPATH, and look into each of the paths.
+		// If the GOPATH hold only one path, there will only be one element in the slice.
+		potentialPaths := getPotentialPaths(goPath, runtime.GOOS)
+		for i, pp := range potentialPaths {
+			goPathProgram := filepath.Join(pp, "bin", program)
+			fileInfo, err := os.Stat(goPathProgram)
+
+			if err != nil && i+1 == len(potentialPaths) {
```

Comment:
This code does the job, but is there a more elegant way to do this check?

Reply:
How does something like:

```
if goPath := os.Getenv("GOPATH"); len(goPath) > 0 {
  // splitGoPath will return paths where to look.
  // Because the GOPATH can take the form of multiple paths (e.g: GOPATH="home/user/go:/usr/local/go")
  // we need to split the GOPATH, and look into each of the paths.
  // If the GOPATH hold only one path, there will only be one element in the slice.
  goPathParts := splitGoPath(goPath, runtime.GOOS)
  for i, pp := range goPathParts {
    goPathProgram := filepath.Join(pp, "bin", program)
    if fileInfo, err := os.Stat(filename); !os.IsNotExist(err) {
      return "", err
    }

    if fileInfo != nil && !fileInfo.Mode().IsDir() {
      logging.V(5).Infof("program %s found in %s/bin", program, pp)
      return goPathProgram, nil
    }
  }

  return "", errors.Wrapf(err, "unable to find program in path: %s", goPath)
}
```

feel to you?

Reply:
I wrote it like this the first time. Let's say GOPATH contains three paths. 
The drawback of doing this: 
```if fileInfo, err := os.Stat(filename); !os.IsNotExist(err)``` is that it will return immediately for the first path that has an error.

I think what we want to do here, is give each path a chance, and return an error (if there is one) only for the last path.

Reply:
Hi @ynden

So the func would only return if the error is NOT IsNotExist - that would mean it is a legit error that we would want to return right?

Reply:
Right! I thought that, even if there was a more serious error (NOT isNotExist), we could try the other paths as well. 
But now I understand, I'll change it, thank you for your guidance!

Reply:
👌